### PR TITLE
E2E cluster network validation

### PR DESCRIPTION
**Pull request by lalon4**
Adding new e2e flow to cover cluster network validation, including DNS domain, Ingress IP and API IP ui validations. 
In addition, added to shared.js new method which saving changes in cluster details page

### CODE REVIEW

File: cypress/integration-qe/e2e-networkValidation.spec.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    saveClusterDetails
+} from './shared';
+  
+import {
+    VALIDATE_CHANGES_TIMEOUT,
+} from './constants';
+
+const VIP_OUT_CIDR = '1.1.1.1'
+const MASTER_IP = ''
+const nicsTableCell = (row, label) =>
```
Comment (ukalifon) [inline cypress/integration-qe/e2e-networkValidation.spec.js:16]: This identifier can be in the shared file

Comment (lalon4) [reply cypress/integration-qe/e2e-networkValidation.spec.js]: fixed

File: cypress/integration-qe/networkValidation.spec.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    cy.get('#form-input-ingressVip-field').type(API_VIP);
+    cy.get('#form-input-ingressVip-field-helper').contains(
+      'Ingress and API IP addresses can not be the same.',
+    );
+  });
+
+  it('check that correct VIP saved', () => {
+    cy.get('#form-input-ingressVip-field').clear();
+    cy.get('#form-input-ingressVip-field').type(INGRESS_VIP);
+    cy.get('button[name="save"]', { timeout: VALIDATE_CHANGES_TIMEOUT }).should(($elem) => {
```
Comment (mareklibra) [inline cypress/integration-qe/networkValidation.spec.js:153]: I believe following is easier:
```
cy.get('button[name="save"]').should('be.disabled')
```

File: cypress/integration-qe/networkValidation.spec.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  it('set invalid IP: same IP as master-0-0', () => {
+    cy.get('#form-input-ingressVip-field').clear().type(MASTER_IP);
+    saveClusterDetails();
+    cy.get('.pf-c-alert').contains('is already in use in cidr');
+    cy.get('.pf-c-alert__action').click();
+  });
+
+  it('save is not allowed due to empty IP', () => {
+    cy.get('#form-input-ingressVip-field').clear();
+    cy.get('button[name="save"]', { timeout: VALIDATE_CHANGES_TIMEOUT }).should(($elem) => {
```
Comment (mareklibra) [inline cypress/integration-qe/networkValidation.spec.js:136]: see bellow

File: cypress/integration/shared.js [hunk 4]
```diff
@@ -191,3 +191,11 @@ export const waitForClusterInstallation = () => {
   // wait up to 1 hour for the progress description to say "Installed"
   cy.contains('div.pf-c-progress__description', 'Installed', { timeout: CLUSTER_CREATION_TIMEOUT });
 };
+
+export const saveClusterDetails = () => {
+  // click the 'save' button in order to save changes in the cluster info
+  cy.get('button[name="save"]', { timeout: VALIDATE_CHANGES_TIMEOUT }).should(($elem) => {
```
Comment (mareklibra) [inline cypress/integration/shared.js:197]: likewise

File: cypress/integration-qe/networkValidation.spec.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+describe('Cluster Ingress IP Validation', () => {
+  it('open cluster details', () => {
+    openCluster(CLUSTER_NAME);
+  });
+
+  it('get master IP', () => {
+    cy.get('#expandable-toggle0').click();
+    cy.get(nicsTable(1, 'IPv4 address')).then(($elem) => {
+      MASTER_IP = $elem.text().split('/')[0];
```
Comment (mareklibra) [inline cypress/integration-qe/networkValidation.spec.js:115]: This works just by accident, please see https://docs.cypress.io/guides/core-concepts/variables-and-aliases.html .

One way to fix it is to use `cy.get().within()`, please grep other tests for examples. Or using `.as()` - not yet used in our tests, IIRC.

File: cypress/integration/shared.js [hunk 6]
```diff
@@ -191,3 +191,11 @@ export const waitForClusterInstallation = () => {
   // wait up to 1 hour for the progress description to say "Installed"
   cy.contains('div.pf-c-progress__description', 'Installed', { timeout: CLUSTER_CREATION_TIMEOUT });
 };
+
+export const saveClusterDetails = () => {
```
Comment (mareklibra) [inline cypress/integration/shared.js:195]: I prefer to pass the `cy` as a parameter here, it will provide flexibility in the future.

**Pull request merged**